File: articles/hashing-indexing-and-retrieval/typescript/src/hashingRetrievalAudit.ts
export type RetrievalStructure = "hash_map" | "database_index" | "inverted_index" | "cache" | "content_addressing";
export type RetrievalCase = { name: string; structure: RetrievalStructure; retrievalQuality: number };
export const cases: RetrievalCase[] = [
  { name: "Article metadata dictionary", structure: "hash_map", retrievalQuality: 84.24 },
  { name: "Case status database index", structure: "database_index", retrievalQuality: 84.20 },
  { name: "Search inverted index", structure: "inverted_index", retrievalQuality: 83.24 },
  { name: "Cache for expensive computations", structure: "cache", retrievalQuality: 82.92 }
];

File: articles/hashing-indexing-and-retrieval/cpp/hashing_retrieval_audit.cpp
#include <iostream>
#include <unordered_map>
int main(){ std::unordered_map<std::string,std::string> index{{"hashing","doc-1|doc-4"}}; std::cout << "case_name,retrieval_quality\nArticle metadata dictionary,84.24\nCase status database index,84.20\nSearch inverted index,83.24\nCache for expensive computations,82.92\n"; }
